lib/spack/spack/test/conftest.py

```diff
@@ -318,6 +318,7 @@ def _skip_if_missing_executables(request):
 @contextlib.contextmanager
 def use_configuration(config):
     """Context manager to swap out the global Spack configuration."""
+    config._get_config_memoized.cache.clear()
```

Comment:
I think we should also discourage use of `spack.config.config = ...` and add a method to explicitly set it in `spack.config`. that method should be called twice here:

```
spack.config.config = config
...
spack.config.config = saved
```

(either way, I think you'll want to clear the memoizations twice)

Reply:
Added `spack.config.replace_config` in 6dcb935